src/createfunctionsjson.ts

```diff
@@ -0,0 +1,302 @@
+                            resultType = getParamType(func.type);
+                            resultDim = getParamDim(func.type);
+                        }
+                    } else {
+                        console.log("No return type specificed. This could be .js filetype, so continue.");
+                    }
+
+                    let result = {
+                        type: resultType,
+                        dimensionality: resultDim
+                    }
+
+                    let options = {
+                        sync: false,
+                        cancelable: isStreamingFunction,
+                        stream: isStreamingFunction
+                    };
+
+                    const funcName = func.name.text;
+                    const metadataItem: ICFVisualFunctionMetadata = {
+                        name: funcName,
+                        id: funcName,
+                        helpurl: getHelpUrl(func),
+                        description,
+                        parameters,
+                        result,
+                        options,
+                    };
+
+                     metadata.push(metadataItem);
+                }
+                else {
+                    //Function was skipped
+                    skippedFunctions.push(func.name.text);
+                }
+            }
+        }
+        ts.forEachChild(node, visit);
+    }
+}
+
+/**
+ * Returns the @helpurl of the JSDo
+ * @param node Node
+ */
+function getHelpUrl(node: ts.Node): string {
+    var helpurl = "";
+    ts.getJSDocTags(node).forEach(
+        (tag: ts.JSDocTag) => {
+            if ((tag.tagName.escapedText as string).toLowerCase() === HELPURL_PARAM) {
+                if (tag.comment) {
+                    helpurl = tag.comment;
+                }
+            }
+        }
+    );
+    return helpurl;
+}
+
+/**
+* This method will parse out all of the @param tags of a JSDoc and return a dictionary
+* @param node - The function to parse the JSDoc params from
+*/
+function getJSDocParams(node: ts.Node): { [key: string]: string } {
+    const jsDocParamInfo = {};
+
+    ts.getAllJSDocTagsOfKind(node, ts.SyntaxKind.JSDocParameterTag).forEach(
+        (tag: ts.JSDocParameterTag) => {
+            if (tag.comment) {
+                const comment = (tag.comment.startsWith('-')
+                    ? tag.comment.slice(1)
+                    : tag.comment
+                ).trim();
+
+                jsDocParamInfo[(tag as ts.JSDocPropertyLikeTag).name.getFullText()] = comment;
+            }
+            else {
+                //Description is missing so add empty string
+                jsDocParamInfo[(tag as ts.JSDocPropertyLikeTag).name.getFullText()] = "";
+            }
+        }
+    );
+
+    return jsDocParamInfo;
+}
+
+/**
+ * Determines if the last parameter is streaming
+ * @param param ParameterDeclaration
+ */
+function isLastParameterStreaming(param?: ts.ParameterDeclaration): boolean {
```

Comment:
The name should just be "isParameterStreaming" as it checks whether the passed in parameter is for the streaming type. The caller determines to check the last parameter.

This should also be used elsewhere to ensure that only the last parameter is streaming. if the user had a streaming function, but added another parameter onto it, they might not realize that they needed to keep the streaming parameter as the last one. Also, you should write tests and include this as a test case.